lametro/models.py

```diff
@@ -191,39 +204,40 @@ def latest_council_seat(self):
         pass
 
     @property
+    def board_office(self):
+
+        try:
+            office_membership = self.memberships\
+                .filter(organization__name=settings.OCD_CITY_COUNCIL_NAME)\
+                .filter(Q(role='Chair') |
+                        Q(role='1st Chair') |
+                        Q(role='2nd Chair') |
+                        Q(role='Vice Chair'))\
+                .get(end_date_dt__gt=Now())
+        except Membership.DoesNotExist:
+            office_membership = None
+
+        return office_membership
+
+    @cached_property
     def committee_sponsorships(self):
         '''
         This property returns a list of ten bills, which have recent actions
         from the organizations that the person has memberships in.
 
         Organizations do not include the Board of Directors.
         '''
-        query = '''
-            SELECT bill_id
-            FROM councilmatic_core_bill as bill
-            JOIN councilmatic_core_action as action
-            ON bill.ocd_id = action.bill_id
-            JOIN councilmatic_core_organization as org
-            ON org.ocd_id = action.organization_id
-            JOIN councilmatic_core_membership as membership
-            ON org.ocd_id = membership.organization_id
-            WHERE membership.person_id='{person}'
-            AND action.date >= membership.start_date
-            AND org.classification = 'committee'
-            ORDER BY action.date DESC
-            LIMIT 10
-        '''.format(person=self.ocd_id)
-
-        with connection.cursor() as cursor:
-            cursor.execute(query)
-            bill_ids = [bill_tup[0] for bill_tup in cursor.fetchall()]
-
-            bills = LAMetroBill.objects.filter(ocd_id__in=bill_ids)
-
-        return bills
-
-
-class LAMetroEventManager(models.Manager):
+        qs = LAMetroBill.objects\
+            .defer('extras')\
+            .filter(actions__organization__classification='committee')\
```

Comment:
it evaluates to the same thing? 

Do we have a style guide for this type of thing?

Reply:
Hmmmmmmmmmm, TIL that chained `filter` calls are _not_ always the same as combining terms in `filter` call. So, chaining filter calls here will return bills with committee actions **and**  bills with actions from current members, not bills with committee actions from current members. 

The language is a little confusing, so here's another example [from the docs](https://docs.djangoproject.com/en/dev/topics/db/queries/#spanning-multi-valued-relationships) and [a StackOverflow thread](https://stackoverflow.com/questions/8164675/chaining-multiple-filter-in-django-is-this-a-bug). 

It seems like you want results to meet both filter conditions, yes? If so, you'll need to combine the terms.

Reply:
This is giving me what I want.

![Screenshot_2019-11-25 Kathryn Barger - Metro Board](https://user-images.githubusercontent.com/536941/69576709-e041a480-0f9a-11ea-8a9f-2fe220d4fdaa.png)

Reply:
combining them into the same select produces the same SQL.

Reply:
That's very mysterious to me. If combining them produces the same SQL, I'd prefer you combine them to avoid the potential for subtle bugs, should the chained method function as advertised in other circumstances – that is:

> Successive filter() calls further restrict the set of objects, but for multi-valued relations, they apply to any object linked to the primary model, not necessarily those objects that were selected by an earlier filter() call.

Reply:
okee doke, done.

Reply:
From the bottom of my superstitious heart, thanks very much.